### PR TITLE
feat: [PL-20872]: Refactor Reference Component

### PR DESCRIPTION
**Pull request by meenakshi-harness**
##### Summary:

Mocks: Old & New 
https://www.figma.com/file/TW2AjvQhWWtRXZNTwa9PrI/Selection-Modals?node-id=2%3A70

##### Screenshots:
Connector Selector component (Github Connector)
Screenshot 2022-01-17 at 7 08 30 PM

Secret Selector Component ( View is for any Entity with less details, hence collapse hidden)
Screenshot 2022-01-17 at 7 08 56 PM

Selected State:
Screenshot 2022-01-17 at 7 09 31 PM

Hover State: 
Screenshot 2022-01-17 at 7 09 46 PM

You can use the following comments to re-trigger PR Checks

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress: `trigger cypress`
- Fix Prettier - `fix prettier`

### CODE REVIEW

**Comment by meenakshi-harness** [comment]
dockerize

File: src/modules/10-common/components/CollapsableList/CollapsableList.module.scss [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+  }
+
+  .hideCollapseIcon {
+    [class*='CollapseHeader--leftSection'] {
+      > span {
+        display: none;
+      }
+    }
+  }
+  .collapseWrapper {
```
[the remark this answers is not in the document]
Comment (meenakshi-harness) [reply src/modules/10-common/components/CollapsableList/CollapsableList.module.scss:49]: These are the classes we send as props to Component. We don't have any css with same name in the Component.

**Comment by meenakshi-harness** [comment]
@abhinavrastogi-harness can you check if this shadow looks good ?
Screenshot 2022-01-27 at 4 54 43 PM

**Comment by meenakshi-harness** [comment]
dockerize

File: src/modules/10-common/components/EntityReference/EntityReference.module.scss [hunk 2]
```diff
@@ -6,83 +6,43 @@
...
-
-  .searchBox {
-    display: flex;
-    .search {
-      flex: 1;
-      > div {
-        > span {
-          > svg {
-            margin-top: 1px;
+    .searchBox {
```
[the remark this answers is not in the document]
Comment (meenakshi-harness) [reply src/modules/10-common/components/EntityReference/EntityReference.module.scss:21]: These are acc. to mocks and also same as previous implementation. Anyways I'll check with on Yuen on this for the search designs and will change this accordingly.

**Comment by meenakshi-harness** [comment]
dockerize

**Comment by meenakshi-harness** [comment]
dockerize

**Comment by meenakshi-harness** [comment]
dockerize

**Comment by meenakshi-harness** [comment]
trigger cypress

**Comment by meenakshi-harness** [comment]
trigger cypress

**Comment by meenakshi-harness** [comment]
retrigger jest

File: src/modules/10-common/components/EntityReference/EntityReference.tsx [hunk 3]
```diff
@@ -133,78 +176,116 @@ export function EntityReference<T>(props: EntityReferenceProps<T>): JSX.Element
...
-      {data.map((item: EntityReferenceResponse<T>) => (
-        <div
-          key={item.identifier}
-          className={cx(css.listItem, recordClassName, {
-            [css.selectedItem]: selectedRecord === item.record
-          })}
-          onClick={() => setSelectedRecord(selectedRecord === item.record ? undefined : item.record)}
-        >
-          {recordRender({ item, selectedScope, selected: selectedRecord === item.record })}
+  const RenderList = () => {
```
Comment (swarajcheguri-harness) [inline src/modules/10-common/components/EntityReference/EntityReference.tsx:248]: optional: why 'R' and not 'r' in RenderList ?

**Review by vardanbansal-harness — DISMISSED** [review]

File: src/modules/30-secrets/components/SecretReference/SecretReference.tsx [hunk 4]
```diff
@@ -224,41 +202,87 @@ const SecretReference: React.FC<SecretReferenceProps> = props => {
...
+                        {item.record.name}
+                      </Text>
+                      {item.record.type === 'SecretText' || item.record.type === 'SecretFile' ? (
+                        <Text lineClamp={1} font={{ size: 'small', weight: 'light' }} color={Color.GREY_600}>
+                          {`${getString('common.ID')}: ${item.identifier}.${
+                            (item.record.spec as SecretTextSpecDTO).secretManagerIdentifier
+                          }`}
+                        </Text>
+                      ) : null}
+                      {item.record.type === 'SSHKey' ? (
```
Comment (mtran7) [inline src/modules/30-secrets/components/SecretReference/SecretReference.tsx:247]: Suggestion: create a map const above for record type for SSHKey, SecretText, SecretFile

Comment (mtran7) [reply src/modules/30-secrets/components/SecretReference/SecretReference.tsx:247]: Can export and use this map from src/modules/30-secrets/modals/CreateOrSelectSecretModal/useCreateOrSelectSecretModal.tsx

const enum SecretTypeEnum {
  SECRET_TEXT = 'SecretText',
  SSH_KEY = 'SSHKey',
  SECRET_FILE = 'SecretFile'
}

File: src/modules/30-secrets/modals/CreateOrSelectSecretModal/useCreateOrSelectSecretModal.tsx [hunk 5]
```diff
@@ -33,21 +47,79 @@ const useCreateOrSelectSecretModal = (
...
+    },
+    {
+      label: getString('secrets.secret.labelFile'),
+      value: 'SecretFile'
+    }
+  ]
+
+  const defaultSecretType = secretTypeOptions.findIndex(val => val.value === props.type)
+  const [secretType, setSecretType] = React.useState<SelectOption>(
+    secretTypeOptions[defaultSecretType === -1 ? 0 : defaultSecretType]
```
Comment (mtran7) [inline src/modules/30-secrets/modals/CreateOrSelectSecretModal/useCreateOrSelectSecretModal.tsx:64]: may need a comment here for why defaultSecretType would be -1. It's not clear for the next person working w/ this file

File: src/modules/30-secrets/modals/CreateOrSelectSecretModal/useCreateOrSelectSecretModal.tsx [hunk 6]
```diff
@@ -33,21 +47,79 @@ const useCreateOrSelectSecretModal = (
...
+  )
+
+  const inputDeps = inputs?.length ? inputs.concat([secretType]) : [secretType]
```
Comment (mtran7) [inline src/modules/30-secrets/modals/CreateOrSelectSecretModal/useCreateOrSelectSecretModal.tsx:67]: would spell out const name. Is it inputDependencies?

File: src/modules/35-connectors/components/ConnectorReferenceField/FormMultiTypeConnectorField.tsx [hunk 7]
```diff
@@ -307,8 +316,17 @@ export const MultiTypeConnectorField = (props: MultiTypeConnectorFieldProps): Re
             placeholder: placeHolderLocal,
             label,
             getString,
-            openConnectorModal
+            openConnectorModal,
+            setPagedConnectorData
           }),
+          disableCollapse: !(type === 'Github'),
```
Comment (mtran7) [inline src/modules/35-connectors/components/ConnectorReferenceField/FormMultiTypeConnectorField.tsx:322]: Is there a reason type Github would disable collapse but not others? Maybe add comment

File: src/modules/30-secrets/components/SecretReference/SecretReference.tsx (hunk 4 above)
Comment (meenakshi-harness) [reply src/modules/30-secrets/components/SecretReference/SecretReference.tsx:247]: Yes we have it already. Forgot to replace this string. Will changes this.

File: src/modules/30-secrets/modals/CreateOrSelectSecretModal/useCreateOrSelectSecretModal.tsx (hunk 6 above)
Comment (meenakshi-harness) [reply src/modules/30-secrets/modals/CreateOrSelectSecretModal/useCreateOrSelectSecretModal.tsx:67]: yes

File: src/modules/35-connectors/components/ConnectorReferenceField/FormMultiTypeConnectorField.tsx (hunk 7 above)
Comment (meenakshi-harness) [reply src/modules/35-connectors/components/ConnectorReferenceField/FormMultiTypeConnectorField.tsx:322]: Yes, inly Github will have collapse and not others. Other connectors will need to onboard this and add details in collapsed view. Will add comment here.

**Comment by meenakshi-harness** [comment]
retrigger jest

**Comment by meenakshi-harness** [comment]
retrigger jest

**Review by mtran7 — APPROVED** [review]

File: src/modules/10-common/components/CollapsableList/CollapsableList.module.scss [hunk 8]
```diff
@@ -0,0 +1,68 @@
...
+  .collapeTitle {
+    min-height: 50px;
+  }
+  .collapseHeader {
+    min-height: 50px;
+    padding-right: var(--spacing-medium) !important;
+    padding-left: var(--spacing-medium) !important;
+    padding-bottom: var(--spacing-small) !important;
+
+    [class*='CollapseHeader--leftSection'] {
```
Comment (sunnykesh-harness) [inline src/modules/10-common/components/CollapsableList/CollapsableList.module.scss:40]: Can not we define style  for just leftSection here?

**Review by sunnykesh-harness — APPROVED** [review]

**Pull request merged**